gwsumm/plot/builtin.py

```diff
@@ -710,15 +710,14 @@ def draw(self, outputfile=None):
 
         # plot
         for ax, arr, pargs in zip(cycle(axes), data, histargs):
-            # set range if not given
-            if pargs.get('range') is None:
-                pargs['range'] = self._get_range(
-                    data,
-                    # use range from first dataset if already calculated
-                    range=histargs[0].get('range'),
-                    # use xlim if manually set (user or INI)
-                    xlim=None if ax.get_autoscalex_on() else ax.get_xlim(),
-                )
+            # set range
+            pargs['range'] = self._get_range(
```

Comment:
Looks like this gets rid of a `if`-statement. Is this intentional and desired?

Reply:
yes, with the extra range arguments, it is not `None`. But the `_get_range()` should give the same result if range is not defined.

Reply:
But wouldn't this impact the whole `TimeSeriesHistogram` class and any child classes? I worry about other classes that may not expect this change. Can you check what else might be impacted.

Reply:
If range is given, and it is the normal 2-d tuple, it `_get_range()` will return this range.

Reply:
@iaraota Sorry, maybe I'm not fully understanding your answer.

Before this PR, if there is no `range` (it is `None`), then there will be a call to `_get_range()` to set the range based on the return from the function. If `range` was provided, then there is no call to `_get_range()`.

With this PR, `_get_range()` is always called no matter what.

I'm asking if this behaviour is intended and have you tested the impacts on `TimeSeriesHistogram` and classes inheriting from this one.

Reply:
@eagoetz the following page has examples using the timeseries histogram with different range parameters: https://ldas-jobs.ligo-la.caltech.edu/~iara.ota/summary/gps/1389301018-1389312018/test_ranges/

In my code, when range is **not** `None`, it calls the `_get_range()` method. However, it only satisfies the condition on line 761 (`if range is not None`). It fails the additional checks introduced in lines 762, 767, and 769 (representing these new options).

It still needs to meet the condition `if range[0] < range[1]` on line 771. This is a **mandatory** requirement for the `hist` function in Matplotlib; violating it will raise a `ValueError` exception.

Reply:
@iaraota Ok, I think I get it now. Thanks for clarifying. The logic in the original code was not very clear but now I think you have improved my understanding. I think this is ok and my concerns about this change